src/main/java/io/github/hapjava/server/impl/json/CharacteristicsController.java

```diff
@@ -59,9 +59,10 @@ public HttpResponse get(HttpRequest request) throws Exception {
       }
     }
     try (ByteArrayOutputStream baos = new ByteArrayOutputStream()) {
-      Json.createWriter(baos)
-          .write(
-              Json.createObjectBuilder().add("characteristics", characteristics.build()).build());
+      JsonWriter writer = Json.createWriter(baos);
+      writer.write(
+          Json.createObjectBuilder().add("characteristics", characteristics.build()).build());
+      writer.close();
```

Comment:
JsonWriter is an AutoClosable, makes more sense to use the try block:

try {(ByteArrayOutputStream baos = new ByteArrayOutputStream();
        JsonWriter writer = Json.createWriter(baos))